Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1

```diff
@@ -280,28 +314,75 @@ function Disable-RemoteDesktopAccess
-                        $output = Remove-AzureServiceRemoteDesktopExtension -ServiceName $serviceName -Slot $_.Slot -Role $_.Role -ErrorAction SilentlyContinue
-                        if($null -eq $output -and $output.OperationStatus -ine "Succeeded")
+                    $_.RemoteAccessDetails | ForEach-Object {
+                        $slotName = $_.Slot
+                        $roleName = $_.Role
+
+                        if ($isEnabledViaExtension)
                         {
-                            $item =  New-Object psobject -Property @{  
-                                CloudServiceName = $serviceName                
-                                ResourceGroupName = $rgName
+                            $output = Remove-AzureServiceRemoteDesktopExtension -ServiceName $serviceName -Slot $slotName -Role $roleName -ErrorAction SilentlyContinue
+                            if($null -eq $output -and $output.OperationStatus -ine "Succeeded")
+                            {
+                                $item = New-Object psobject -Property @{
+                                    CloudServiceName = $serviceName
+                                    ResourceGroupName = $rgName
+                                    SlotName = $slotName
+                                    RoleName = $roleName
+                                }
+
+                                $isServiceRemediated = $false
+                                $skippedCloudServiceFromRemediation += $item
+                                break;
+                            }
+                        }
+                        else
+                        {
+                            $index = 0
+                            $slotConfiguration=[xml]$_.slotConfiguration
+                            $slotConfiguration.ServiceConfiguration.Role | % {
+                                if ($($_.name) -eq $roleName)
+                                {
+                                    # The configuration (XML) retrieved from the Get-AzureDeployment command will be used in the call to Set-AzureDeployment, but, with Remote Access set to false.
+                                    $remoteAccessSetting = $slotConfiguration.ServiceConfiguration.Role[$index].ConfigurationSettings.Setting | Where-Object { $_.name -eq "Microsoft.WindowsAzure.Plugins.RemoteAccess.Enabled" }
+                                    $remoteAccessSetting.value = $false
+
+                                    $configurationFile = "$($folderpath)\$serviceName_$slotName_$roleName.cfg"
+
+                                    # Create a temporary file to store this configuration and upload this file to the Cloud Service instance.
+                                    $slotConfiguration.Save($configurationFile)
+
+                                    # TODO: Explore optimizing calls to Set-AzureDeployment by updating all eligible Roles in a Deployment Slot in one shot.
```

Comment:
I think better would be to add task in backlog and not put TODO comments in publicly available file. Thoughts?

Reply:
Will track this internally.